test/it/helpers/browser.js

```diff
@@ -1,113 +1,206 @@
+    var enabled = false
+    var error = null
+    try {
+      if (window && window.localStorage) {
+        const key = Math.random().toString()
+        window.localStorage.setItem(key, key)
+        enabled = window.localStorage.getItem(key) === key
+        window.localStorage.removeItem(key)
+      }
+    } catch (e) {
+      error = e
+    }
+    return [error, enabled]
+  })
+  const error = result[0]
+  const enabled = result[1]
+  if (error) {
+    console.warn(`Error while probing localstorage: ${JSON.stringify(error)}`)
+  }
+  if (!enabled) {
+    console.warn('localstorage not supported')
   }
+  return enabled
 }
 
-export function deleteAllCookies () {
-  try {
-    browser.execute(() => {
-      try {
-        const cookies = document.cookie.split('; ')
-        for (let c = 0; c < cookies.length; c++) {
-          const d = window.location.hostname.split('.')
-          while (d.length > 0) {
-            const cookieBase = encodeURIComponent(cookies[c].split(';')[0].split('=')[0]) + '=; expires=Thu, 01-Jan-1970 00:00:01 GMT; domain=' + d.join('.') + ' ;path='
-            const p = location.pathname.split('/')
-            document.cookie = cookieBase + '/'
-            while (p.length > 0) {
-              document.cookie = cookieBase + p.join('/')
-              p.pop()
-            }
-            d.shift()
+export async function deleteAllCookies () {
+  const error = await browser.execute(function () {
+    try {
+      const cookies = document.cookie.split('; ')
+      for (let c = 0; c < cookies.length; c++) {
+        const d = window.location.hostname.split('.')
+        while (d.length > 0) {
+          const cookieBase = encodeURIComponent(cookies[c].split(';')[0].split('=')[0]) + '=; expires=Thu, 01-Jan-1970 00:00:01 GMT; domain=' + d.join('.') + ' ;path='
+          const p = location.pathname.split('/')
+          document.cookie = cookieBase + '/'
+          while (p.length > 0) {
+            document.cookie = cookieBase + p.join('/')
+            p.pop()
           }
+          d.shift()
         }
-      } catch (ex) {
-        console.error(ex)
       }
-    })
-  } catch (e) {
-    console.error('Error while Deleting all cookies', e)
+      return null
+    } catch (e) {
+      return e
+    }
+  })
+  if (error) {
+    console.error(`failed cleaning cookies: ${error}`)
   }
 }
 
 export function isMobileSafari () {
-  return browser.capabilities.browserName === 'safari' && browser.capabilities.real_mobile
+  return browser.capabilities.browserName === 'safari' &&
+    (browser.capabilities.realMobile || browser.capabilities.real_mobile)
 }
 
 export function isMobileSafari14OrNewer () {
   return browser.capabilities.browserName === 'safari' &&
-    parseInt(browser.capabilities.version.substring(0, 2)) >= 14 &&
-    browser.capabilities.real_mobile
+    (
+      (browser.capabilities.browserVersion && parseInt(browser.capabilities.browserVersion.substring(0, 2)) >= 14) ||
```

Comment:
These are a bit annoying now as they can be reported either w3c or in json web format.
Sadly the wdio heuristic (exposed via `browser.isW3C`) for deciding whether w3c is used is unreliable, so we have to check for both